tests/ttnn/unit_tests/operations/ccl/perf/perf_csv.py

```diff
@@ -52,70 +57,147 @@ def clean_attributes(attributes):
     ]
     filtered_df[numeric_columns] = filtered_df[numeric_columns].apply(pd.to_numeric, errors="coerce")
 
     def calculate_min_avg_max_by_common_runs(df):
-        group_columns = ["ATTRIBUTES_BACKUP", "Input Shape", "dim", "num_links", "topology", "Layout", "Data Type"]
+        group_columns = [
+            "ATTRIBUTES_BACKUP",
+            "Input Shape",
+            "Output Shape",
+            "dim",
+            "num_links",
+            "topology",
+            "Layout",
+            "Data Type",
+            "OP CODE",
+        ]
         results = []
 
         for name, group in df.groupby(group_columns):
-            group_excluded = group.iloc[4:]
+            n_chips = group["n_chips"].iloc[0]
+            group_excluded = group.iloc[2 * n_chips :]
             if not group_excluded.empty:
                 min_values = group_excluded[numeric_columns].min().round(2)
                 avg_values = group_excluded[numeric_columns].mean().round(2)
                 max_values = group_excluded[numeric_columns].max().round(2)
 
+                longest_device_fw_time = max_values["DEVICE FW DURATION [ns]"]
+                longest_erisc_fw_time = max_values["DEVICE ERISC KERNEL DURATION [ns]"]
+
                 result_row = {
                     **{col: f"{min_values[col]} - {avg_values[col]} - {max_values[col]}" for col in numeric_columns},
+                    "longest_device_fw_time": longest_device_fw_time,
+                    "longest_erisc_fw_time": longest_erisc_fw_time,
                     **{key: value for key, value in zip(group_columns, name)},
+                    "n_chips": n_chips,
                 }
                 results.append(result_row)
 
         return pd.DataFrame(results)
 
     average_values_by_common_run = calculate_min_avg_max_by_common_runs(filtered_df)
 
-    print_order = ["Input Shape", "dim", "num_links", "topology", "Layout", "Data Type"] + numeric_columns
-    average_values_by_common_run = average_values_by_common_run[print_order]
+    def calculate_bandwidth(row):
+        longest_device_fw_time = row["longest_device_fw_time"]
+        longest_erisc_fw_time = row["longest_erisc_fw_time"]
+
+        input_tensor_volume = (
+            int(row["Input Shape"].split(",")[0][1:])
+            * int(row["Input Shape"].split(",")[1])
+            * int(row["Input Shape"].split(",")[2])
+            * int(row["Input Shape"].split(",")[3][:-1])
+        )
+
+        output_tensor_volume = (
+            int(row["Output Shape"].split(",")[0][1:])
+            * int(row["Output Shape"].split(",")[1])
+            * int(row["Output Shape"].split(",")[2])
+            * int(row["Output Shape"].split(",")[3][:-1])
+        )
+
+        op_bw, link_bw = None, None
+        n_chips = row["n_chips"]
+        if row["topology"] == "Ring":
+            if row["OP CODE"] == "AllGather":
+                op_bw = (output_tensor_volume * (n_chips - 1) / n_chips) / longest_device_fw_time
```

Comment:
It looks like the element size is missing here because when I manually calculate with the worst case FW times, I expect a number twice as high as what I see for BW calcs.